config/application.rb

```diff
@@ -20,46 +25,29 @@ class Application < Rails::Application
     # config.time_zone = "Central Time (US & Canada)"
     # config.eager_load_paths << Rails.root.join("extras")
 
-    config.eager_load_paths << Rails.root.join('lib/')
+    # Local customisations
+    config.autoload_lib(ignore: %w(assets tasks generators))
     # Pull in folders without namespacing
-    config.eager_load_paths << Rails.root.join('app', 'models', 'areas')
-
-    # Rails generators are auto-discovered from fixed locations. The paths dont match the
-    # expected module structure. However when eager loading is enabled Zeitwerk refuses to
-    # load the code and the app fails to boot.
-    #
-    # Rails 7 has a solution to this: https://github.com/rails/rails/pull/48572
-    #
-    # For now, explicitly ignore the generator from being auto-loaded.
-    Rails.autoloaders.main.ignore(Rails.root.join('lib/generators/comparison_report/comparison_report_generator.rb'))
-
-    #config.time_zone = 'London'
-    # optional - note it can be only :utc or :local (default is :utc)
-    # HAS to be UTC for group by date to work
-    config.active_record.default_timezone = :utc
-
+    config.eager_load_paths << Rails.root.join('app/models/areas')
     # For our application date helpers to use to optionally display times in configured zone
     config.display_timezone = 'London'
-
     config.middleware.use Rack::Attack
     config.middleware.use Rack::XRobotsTag
-
     # uploaded SVG files are served as octet stream by default for security
     # this will remove them from the list of binary file types, but is a slight risk
     config.active_storage.content_types_to_serve_as_binary.delete("image/svg+xml")
-
     # session cookie config will be overridden in production.rb
     config.session_store :cookie_store, key: '_energy-sparks_session'
-
     config.after_initialize do
       if EnergySparks::FeatureFlags.active?(:use_site_settings_current_prices)
         BenchmarkMetrics.set_current_prices(prices: SiteSettings.current_prices)
       end
-
+      # https://stackoverflow.com/questions/77366033/allow-actiontext-tags-in-rails-7-1-with-new-sanitizers
```

Comment:
what do we need to test to ensure this is working? Is it just image and other embeds in trix content?

Reply:
This was added in https://github.com/Energy-Sparks/energy-sparks/commit/e9c77fcdf96de1ad2f2325e37a43bd02d6cf4860#diff-50fc7bad8921b4b92c8b7baa63ae23af079068f87df641c94e6e7909b75deba0 - do charts relate to actiontext?

Reply:
We embed charts in the "school specific description" fields for activities. This is a version of the instructions that includes charts that show the actual data for the school, so when they view the activity type, or their recorded activity, the charts are displayed. 

The charts are embedded in text using a trix extension. What's actually stored is a Mustache template (see `school_template_spec.rb`). But this gets processed to replace it with a chart tag. Looks like this config is avoiding ActionText stripping the attributes when the content is actually rendered. 

I've checked that these embedded charts are still rendering on test: https://test.energysparks.uk/schools/stepaside-primary-school/activities/2320